include/sys/zio_compress.h

```diff
@@ -51,7 +51,32 @@ enum zio_compress {
 	ZIO_COMPRESS_GZIP_9,
 	ZIO_COMPRESS_ZLE,
 	ZIO_COMPRESS_LZ4,
-	ZIO_COMPRESS_FUNCTIONS
+	ZIO_COMPRESS_FUNCTIONS,
+	ZIO_COMPRESS_QOS_10,
+	ZIO_COMPRESS_QOS_20,
+	ZIO_COMPRESS_QOS_30,
+	ZIO_COMPRESS_QOS_40,
+	ZIO_COMPRESS_QOS_50,
+	ZIO_COMPRESS_QOS_100,
+	ZIO_COMPRESS_QOS_150,
+	ZIO_COMPRESS_QOS_200,
+	ZIO_COMPRESS_QOS_250,
+	ZIO_COMPRESS_QOS_300,
+	ZIO_COMPRESS_QOS_350,
+	ZIO_COMPRESS_QOS_400,
+	ZIO_COMPRESS_QOS_450,
+	ZIO_COMPRESS_QOS_500,
+	ZIO_COMPRESS_QOS_550,
+	ZIO_COMPRESS_QOS_600,
+	ZIO_COMPRESS_QOS_650,
+	ZIO_COMPRESS_QOS_700,
+	ZIO_COMPRESS_QOS_750,
+	ZIO_COMPRESS_QOS_800,
+	ZIO_COMPRESS_QOS_850,
+	ZIO_COMPRESS_QOS_900,
+	ZIO_COMPRESS_QOS_950,
+	ZIO_COMPRESS_QOS_1000,
+	ZIO_COMPRESS_META_FUNCTIONS
```

Comment:
Unless we are introducing new compression algorithms, we probably should not use up this address space. I'd rather see you add a zfs property called bandwidth where this can be set. Then have it take effect when compression=on is set.

Also, this does not play well with inheritance. It is too easy for a system administrator to think that bandwidth of all children would be limited by their parent's bandwidth (maybe this should be done?). Having a separate bandwidth property that is not inherited would make it more clear that the children do share bandwidth with their parents.